src/browser/setupWorker/start/createResponseListener.ts

```diff
@@ -25,10 +25,14 @@ export function createResponseListener(context: SetupWorkerInternalContext) {
       return
     }
 
-    const response =
-      responseJson.status === 0
-        ? Response.error()
-        : new Response(responseJson.body, responseJson)
+    let response: Response
+    if (responseJson.status === 0) {
+      response = Response.error()
+    } else if (isResponseWithoutBody(responseJson.status)) {
+      response = new Response(null, responseJson)
+    } else {
+      response = new Response(responseJson.body, responseJson)
+    }
```

Comment:
Handling this here instead of in the worker. this gives us a more integrated place to handle this, and makes it easy to reuse the helper from interceptors

Reply:
could extract to a small method, not sure it's worth jumping out of line here though